flake.nix

```diff
@@ -0,0 +1,52 @@
+{
+  description = "tyrian-dev";
+
+  inputs = {
+    nixpkgs.url = github:nixos/nixpkgs/nixpkgs-unstable;
+    flake-utils.url = github:numtide/flake-utils;
+    flake-compat = {
+      url = github:edolstra/flake-compat;
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, ... }:
+    let
+      supportedSystems = [ "aarch64-darwin" "aarch64-linux" "x86_64-linux" ];
```

Comment:
Running: `nix develop .#js` (<-- if that's valid, see my other comment)

Gives the following error:

```
error: flake 'git+file:///(...)/tyrian' does not provide attribute 'packages.x86_64-darwin.js', 'legacyPackages.x86_64-darwin.js' or 'js'
```

I believe I've installed flakes correctly, but maybe not! Any ideas?

Reply:
You can try now :)

Reply:
This also turned out to be because I was on a slightly older version of Nix - works now.

Reply:
Ah yes, it requires Nix 2.4 at least, which is when flakes were introduced. Still, with older versions you can use the old `nix-shell` as per the instructions on the README file.

Reply:
Yes indeed, the old shell command worked perfectly which made me suspicious that it was at least partially my system. I was running a pre-release of 2.4 somehow.